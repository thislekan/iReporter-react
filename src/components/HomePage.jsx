import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import style from '../styles/HomePage.css';
import phoneImage from '../media/phone-1682317_1280.png';
import screenshot from '../media/shot.png';
import image1 from '../media/ryan-tang-281277-unsplash.jpg';
import megaphone from '../media/megaphone.jpg';

function HomePage(props) {
  const { history } = props;
  return (
    <div className={style.homepage}>
      <section className={style.welcome}>
        <div className={style.paddedBody}>
          <div className={style.welcomeText}>
            <h1 className={style.homepage__intro}>You too can help
            <span>  </span>
              <span className={style.endText}>end</span> corruption.
          </h1>
            <button
              onClick={() => history.push('/signup')}
              id="go-to-signup"
            >
              Sign Up
            </button>
          </div>
          <div className={style.welcomeInvite}>
            <div className={style.imgDiv}>
              <img src={phoneImage} alt="transparent phone" className={style.parentImage} />
              <img src={screenshot} alt="report incident screenshot" className={style.childImage} />
            </div>
          </div>
        </div>
      </section>
      <section className={style.description}>
        <div className={style.paddedBody}>
          <div className={style.intro}>
            <h3>Lend the government your voice</h3>
          </div>
          <div className={style.descriptionBody}>
            <div className={style.imgDiv}>
              <img src={megaphone} alt="protester with megaphone" className={style.descriptionBodyImage} />
            </div>
            <div className={cx(style.descriptionBodyText, style.textDiv)}>
              <p>With an internet connection, you too can help fight corruption.
                All you need to do is register with us and report incidents with their location,
              leaving the rest to us</p>
              <p>We make sure you remain anonymous
                while we investigate and you get real
              time notifications on your reorted incidents.</p>
            </div>
          </div>
        </div>
      </section>
      <section className={style.process}>
        <div className={style.paddedBody}>
          <div className={cx(style.processIntro, style.intro)}>
            <h3>How it works.</h3>
          </div>
          <div className={style.processBody}>
            <div className={style.imgDiv}>
              <img src={image1} alt="" />
            </div>
            <div className={cx(style.processBodyText, style.textDiv)}>
              <p className={style.subHeader}>You too can fight corruption in 3 easy steps</p>
              <div>
                <p><span className={style.numbers}>1.</span> Sign up and verify your email.</p>
                <p><span className={style.numbers}>2.</span> Login after verification.</p>
                <p><span className={style.numbers}>3.</span> Report an incident.</p>
              </div>
            </div>
          </div>
        </div>
      </section>
      <section className={style.stories}>
        <div className={style.paddedBody}>
          <div className={style.intro}>
            <h3>Our numbers have been good</h3>
          </div>
          <div className={style.storiesBreakdown}>
            <div className={style.storiesBreakdownDetails}>
              <div className={style.incident}>
                <span>All inicdents</span>
                <span className={cx(style.incidentStats, style.all)}>1000</span>
              </div>
              <div className={style.incident}>
                <span>Resolved incidents</span>
                <span className={cx(style.incidentStats, style.resolved)}>550</span>
              </div>
              <div className={style.incident}>
                <span>Rejected incidents</span>
                <span className={cx(style.incidentStats, style.rejected)}>200</span>
              </div>
              <div className={style.incident}>
                <span>Under Investigation</span>
                <span className={cx(style.incidentStats, style.investigating)}>250</span>
              </div>
            </div>
            <div className={style.textDiv}>
              <p>We pride ourselves in the numbers of resolved incidents,
                but we aim for even better results.
              Our team of investigators are thorough and discreet in their approach.</p>
              <p>Every incident is treated as a priority.
                Our system and approach are void of bias,
                meaning we hold ourselves to a great standard.
              The numbers tell our stories for us.</p>
            </div>
          </div>
        </div>
      </section>
      <footer>
        <div className={style.disclaimer}>
          <p>disclaimer</p>
          <p>iReporter is a non governmental organization
          committed to ending corruption in nigeria</p>
        </div>
        <div>
          <p>contact us <a href="mailto:email@ireporter.net">email@ireporter.net</a></p>
        </div>
      </footer>
    </div>
  );
}

HomePage.propTypes = {
  history: PropTypes.object.isRequired,
};

export default HomePage;
