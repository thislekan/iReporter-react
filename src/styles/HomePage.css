.paddedBody{
    padding: 0 1rem;
    position: relative;
}
.welcome{
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-evenly;
    background: rgba(239, 239, 239, 0.5);
    height: 100vh;
}
.welcomeText{text-align: center;}
.welcomeText button{
    height: 5rem;
    width: 15rem;
    border: 1px solid #117d5a;
    background: #117d5a;
    border-radius: .25rem;
    font-size: 1.3rem;
    font-weight: 600;
    color: white;
    margin: -1rem 0;
}
.imgDiv{
    position: relative;
    display: flex;
    align-items: center;
    justify-content: center;
}
.parentImage{
    height: 350px;
}
.childImage{
  position: absolute;
  height: calc(100% - 13.5rem);
  width: calc(100% - 177px);
  border: 1px solid transparent;
  border-radius: .5rem;
  margin-top: .25rem;
}
.homepage__intro{
  margin: 15px;
  font-size: 2.5rem;
  white-space: pre-line;
}
.endText{
    color: #cf4241;
}
.homepage section{
  /* height: 100vh; */
  text-align: center;
}
.description{
    height: 115vh;
}
.process{
  height: 130vh;
  background-color: #f7f7f7;
}
.processBodyText{
    height: 300px;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
}
.descriptionBodyText {
    height: 50%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-evenly;
}
/* #description{
    text-align: center;
} */
.intro{
    margin: 1rem 0;
    font-size: 2rem;
    color: #1d394a;
}
.descriptionBody,
.processBody{
    display: flex;
    flex-direction: column-reverse;
    align-items: center;
    justify-content: space-evenly;
}
.descriptionBodyImage{
    height: 250px;
    width: 300px;
    object-fit: scale-down;
    object-position: center;
}
.textDiv{
    font-size: 1.5rem;
    letter-spacing: 1px;
    text-align: left;
    /* margin: 0; */
    /* padding: 0 1rem; */
}
.processIntro h3{
    padding-top: 1.5rem;
    margin: 0;
}
.subHeader{
    font-size: 1.6rem;
    margin: 0;
    text-align: center;
    color: #23948c;
}
/* .process-body{
    flex-direction: column;
} */
.processBody img{
    height: 300px;
    border-radius: .25rem;
    border: 1px solid rgba(29, 57, 74, 0.3);
    box-shadow: 0 0 0 -4px #f7f7f7;
}
.process-body__text{
    /* margin: 2rem 0; */
    padding: 0 1rem;
}
.numbers{
    color: #cf4241;
    margin-right: 1rem;
}
.storiesBreakdownDetails{
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-gap: 1rem;
    justify-items: center;
}
.incident{
    height: 10rem;
    width: 10rem;
    border: 1px solid rgba(29, 57, 74, 0.1);
    border-radius: 100%;
    display: flex;
    align-items: center;
    justify-content: space-evenly;
    flex-direction: column;
    font-size: 1.3rem;
    font-weight: 600;
    letter-spacing: 1px;
    word-break: break-word;
}
.incidentStats{
    font-size: 1.65rem;
}
.resolved{
    color: #117d5a;
}
.rejected{
    color: #cf4241
}
.all{
    color: #f2b21c;
}
.investigating{
    color: #1d394a;
}
footer{
    height: 30vh;
    background: #1d394a;
    color: white;
    text-align: center;
    padding: 3rem;
    font-size: 1.5rem;
    display: flex;
    align-items: center;
    justify-content: space-evenly;
    flex-direction: column;
}
#disclaimer p{
    white-space: pre-line;

}
#disclaimer span{
    color: white;
    margin-top: 1rem;
}
@media screen and (min-width: 330px) and (max-width: 375px){
  .childImage{
    width: calc(100% - 230px);
  }
}
@media screen and (min-width: 377px) and (max-width: 380px){
  .childImage{
    width: calc(100% - 235px);
  }
}
@media screen and (min-width: 387px) and (max-width: 425px){
  .childImage{
    width: calc(100% - 265px);
  }
}
@media screen and (min-width: 700px){
  .welcome .paddedBody{
      display: grid;
  grid-template-columns: 1fr 1fr;
  align-items: center;
  }
  .description,
  .process,
  .stories{
      height: 50vh;
  }
  .paddedBody{
      padding: 0 5rem;
      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: center;
      height: 100%;
  }
  .descriptionBody, 
  .processBody,
  .storiesBreakdown{
      display: grid;
      grid-template-columns: 1fr 1fr;
      grid-gap: 3rem;
  }
  .subHeader{
      text-align: left;
  }
  .processBody img{
      height: 200px;
      margin-top: 1rem;
  }
  .homepage__intro{
    width: 27rem;
  }
  footer{
      height: 15vh;
      padding: 0 5rem;
  }
}
@media screen and (min-width: 900px){
  .welcome .paddedBody{
    grid-column-gap: 10rem;
  }
  .paddedBody{
    padding: 0 15rem;
  }
  footer{
    height: 30vh;
    display: grid;
    grid-template-columns: 1fr 1fr;
    padding: 0 10rem;
  }
}