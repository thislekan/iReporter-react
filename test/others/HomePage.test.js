import React from 'react';
import { shallow } from 'enzyme';
import HomePage from '../../src/components/HomePage.jsx';

const push = jest.fn();
describe('<HomePage />', () => {
  it('should render the homepage', () => {
    const wrapper = shallow(<HomePage history={{ push }} />);
    expect(wrapper).toMatchSnapshot();
    const btn = wrapper.find('#go-to-signup');
    btn.simulate('click', f => f);
    expect(push).toHaveBeenCalled();
  });
});
