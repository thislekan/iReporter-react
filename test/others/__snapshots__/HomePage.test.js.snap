// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render the homepage 1`] = `
<div>
  <section>
    <div>
      <div>
        <h1>
          You too can help
          <span>
              
          </span>
          <span>
            end
          </span>
           corruption.
        </h1>
        <button
          id="go-to-signup"
          onClick={[Function]}
        >
          Sign Up
        </button>
      </div>
      <div>
        <div>
          <img
            alt="transparent phone"
            src={Object {}}
          />
          <img
            alt="report incident screenshot"
            src={Object {}}
          />
        </div>
      </div>
    </div>
  </section>
  <section>
    <div>
      <div>
        <h3>
          Lend the government your voice
        </h3>
      </div>
      <div>
        <div>
          <img
            alt="protester with megaphone"
            src={Object {}}
          />
        </div>
        <div
          className=""
        >
          <p>
            With an internet connection, you too can help fight corruption. All you need to do is register with us and report incidents with their location, leaving the rest to us
          </p>
          <p>
            We make sure you remain anonymous while we investigate and you get real time notifications on your reorted incidents.
          </p>
        </div>
      </div>
    </div>
  </section>
  <section
    className={[Function]}
  >
    <div>
      <div
        className=""
      >
        <h3>
          How it works.
        </h3>
      </div>
      <div>
        <div>
          <img
            alt=""
            src={Object {}}
          />
        </div>
        <div
          className=""
        >
          <p>
            You too can fight corruption in 3 easy steps
          </p>
          <div>
            <p>
              <span>
                1.
              </span>
               Sign up and verify your email.
            </p>
            <p>
              <span>
                2.
              </span>
               Login after verification.
            </p>
            <p>
              <span>
                3.
              </span>
               Report an incident.
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
  <section>
    <div>
      <div>
        <h3>
          Our numbers have been good
        </h3>
      </div>
      <div>
        <div>
          <div>
            <span>
              All inicdents
            </span>
            <span
              className=""
            >
              1000
            </span>
          </div>
          <div>
            <span>
              Resolved incidents
            </span>
            <span
              className=""
            >
              550
            </span>
          </div>
          <div>
            <span>
              Rejected incidents
            </span>
            <span
              className=""
            >
              200
            </span>
          </div>
          <div>
            <span>
              Under Investigation
            </span>
            <span
              className=""
            >
              250
            </span>
          </div>
        </div>
        <div>
          <p>
            We pride ourselves in the numbers of resolved incidents, but we aim for even better results. Our team of investigators are thorough and discreet in their approach.
          </p>
          <p>
            Every incident is treated as a priority. Our system and approach are void of bias, meaning we hold ourselves to a great standard. The numbers tell our stories for us.
          </p>
        </div>
      </div>
    </div>
  </section>
  <footer>
    <div>
      <p>
        disclaimer
      </p>
      <p>
        iReporter is a non governmental organization committed to ending corruption in nigeria
      </p>
    </div>
    <div>
      <p>
        contact us 
        <a
          href="mailto:email@ireporter.net"
        >
          email@ireporter.net
        </a>
      </p>
    </div>
  </footer>
</div>
`;
